src/nifgen/system_tests/test_system_nifgen.py

```diff
@@ -245,3 +245,52 @@ def test_reset_with_default(session):
     assert non_default_arb_sample_rate == 100000000.0
     session.reset_with_defaults()
     assert session.arb_sample_rate == 250000000.0
+
+
+def test_named_waveform_operations(session):
+    wfm_name = 'Waveform'
+    wfm_size = 4096
+    write_offset = 0
+    wfm_data_f64 = [0.0, 1.0]
+    wfm_data_i16 = [1, 0]
+    session.allocate_named_waveform(wfm_name, wfm_size)
+    session.set_named_waveform_next_write_position(wfm_name, nifgen.RelativeTo.START, write_offset)
+    session.write_named_waveform_f64(wfm_name, wfm_data_f64)
+    session.write_named_waveform_i16(wfm_name, wfm_data_i16)
+    session.delete_named_waveform(wfm_name)
+
+
+def test_adjust_sample_clock_relative_delay():
+    with nifgen.Session('', False, 'Simulate=1, DriverSetup=Model:5421;BoardType:PXI') as session:  # 5433 is not supporting adjust_sample_clock_relative_delay right now
+        delay = 1e-09
+        session.adjust_sample_clock_relative_delay(delay)
+
+
+def test_wait_until_done(session):
+    session.wait_until_done(20)
+
+
+def test_user_standard_waveform(session):
+    wfm_points = [1] * 8192
+    session.output_mode = nifgen.OutputMode.FUNC
+    session.configure_standard_waveform(nifgen.Waveform.USER, 1.0, 2000000, 1.0, 0.0)
+    session.define_user_standard_waveform(wfm_points)
+    session.clear_user_standard_waveform()
+
+
+# TODO(bhaswath): Doesn't work, issue #596
```

Comment:
Did you try my suggestion?
https://github.com/ni/nimi-python/issues/596#issuecomment-344280083

Reply:
Yes, I tried and it did not throw a driver error.
Have updated in the issue

Reply:
@texasaggie97 got to the bottom of it, we'll have to tackle it separately.